ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java

```diff
@@ -119,4 +162,291 @@ a new Eth1DataVote(eth1_data=block.eth1_data, vote_count=1).
       votes.add(new Eth1DataVote(block.getEth1_data(), UnsignedLong.ONE));
     }
   }
+
+  /**
+   * @param state
+   * @param block
+   */
+  public static void proposer_slashing(BeaconState state, BeaconBlock block) {
+    assert block.getBody().getProposer_slashings().size() <= MAX_PROPOSER_SLASHINGS;
+
+    for (ProposerSlashing proposer_slashing : block.getBody().getProposer_slashings()) {
+      Validator proposer =
+          state.getValidator_registry().get(proposer_slashing.getProposer_index().intValue());
+
+      assert proposer_slashing
```

Comment:
assert shouldn't be used outside test code (it will always return true unless a special build flag is used). I suggest using this instead: import static com.google.common.base.Preconditions.checkArgument.  I have been replacing all asserts with checkArgument.

Reply:
Ok thanks :)